src/api.js

```diff
@@ -79,48 +79,50 @@ api.declare({
     '**Warning** this api end-point is **not stable**.',
   ].join('\n'),
 }, async function(req, res) {
- 
   let {namespace} = req.params;
+  let newNamespace = await setNamespace(this, namespace);
+  
+  res.reply({
+    namespace: newNamespace.namespace,
+    username: newNamespace.username,
+    password: newNamespace.password,
+  });
+});
 
-  //check for any entries that contain the requested namespace
-  let data = await this.Namespaces.query({
-    namespace:          this.Namespaces.op.equal(namespace),
+/* Retrieve any entries containing the requested namespace.
+ * If an entry exists, use it. Otherwise, create a new entry and an associated rabbit user.*/
+async function setNamespace(context, namespace) {
+  
+  let data = await context.Namespaces.query({
+    namespace:     context.Namespaces.op.equal(namespace),
   }, {
-    limit:            250, 
-  }
-  );
-
-  var newNamespace;
-
-  if (data.entries.length === 0) {
-    //create a new entry if none exists 
-    
-    newNamespace = await this.Namespaces.create({
-      namespace: namespace,
-      username: slugid.v4(),
-      password: slugid.v4(),
-      created:  new Date(),
-      expires:  taskcluster.fromNow('1 day'),
-    });
-
-    await this.rabbit.createUser(newNamespace.username, newNamespace.password, ['taskcluster-pulse']);
+    limit:         250, 
+  });
 
-  } else if (data.entries.length === 1) { 
-    //if a namespace already exists, use the loaded username & password
+  let newNamespace;
+  let entryDNE = data.entries.length === 0;
+  
+  if (entryDNE) {
```

Comment:
Habit, I suppose - I usually try and give a descriptive name to the condition. Can fix that if it's not necessary though.